src/library/pagination/pagination.component.js

```diff
@@ -0,0 +1,170 @@
+// Import Template
+import template from './pagination.html';
+
+class controller {
+  constructor($log, $scope) {
+    Object.assign(this, {
+      $log,
+      $scope
+    });
+
+    this.currentPage = 0; // binding
+    this.disabled = false; // binding
+    this.itemsPerPage = 0; // binding
+    this.size = ''; // binding
+    this.totalItems = 0; // binding
+    this.visiblePageBuffer = 0; // binding
+
+    this.pages = [];
+    this.totalPages = [];
+  }
+
+  $onInit() {
+    // provide defaults
+    this.currentPage = this.currentPage || 1;
+    this.visiblePageBuffer = this.visiblePageBuffer || this.visiblePageBuffer === 0 ? this.visiblePageBuffer : 3;
+    this.size = this.size || '';
+
+    // validate bindings
+    if (typeof this.itemsPerPage !== 'number' || isFinite(this.itemsPerPage) === false) {
+      this.$log.error('invalid ngbsPagination::itemsPerPage: ', this.itemsPerPage, 'expecting a number');
+    }
+
+    if (this.size && (typeof this.size !== 'string' || ['', 'lg', 'sm'].includes(this.size) === false)) {
+      this.$log.error('invalid ngbsPagination::size: ', this.itemsPerPage, 'expecting sm or lg');
+    }
+
+    if (typeof this.totalItems !== 'number' || isFinite(this.totalItems) === false) {
+      this.$log.error('invalid ngbsPagination::totalItems: ', this.totalItems, 'expecting a number');
+    }
+
+    this.$scope.$watch(() => `${this.currentPage}${this.itemsPerPage}${this.totalItems}${this.visiblePageBuffer}`, () => {
```

Comment:
why no take advantage of the wonderful [lifecycle hooks components](https://docs.angularjs.org/guide/component#component-based-application-architecture) come with? specifically $onChanges(changesObj). Not obligatory, just a suggestion...